### ISSUE TITLE
feature request: support for IExplorer

### ISSUE THREAD

**Issue by ccasalengua** [wakirin/Lightpick#67]
Hi, would it be possible to add support for iexplorer (at least IE11 +) (run for windows 7)

I have seen 2 things fail: assing and startsWith. I try several polyfill but it has not worked.

**Comment by amcsi** [comment]
Also when using postcss autoprefixer's grid autoplacement feature, it complains about grid-template-rows not explicitly being given in `lightpick.css`.

**Comment by luckyexpert** [comment]
Yes! i was just about committed to adding this to my app but with no IE support makes it a no go. W3Counter says 3.25% of users are using IE11, so 3 out of every 100 visitors unable to use the datepicker would not be good. Sure hope this can get fixed!

**Comment by jeffkarney** [comment]
I would argue that it is a waste of effort to support any version of IE before Edge or any browser below 20% (think the 80/20 rule). These people will never change unless they are forced to. The only way to do that is to stop supporting them.

Now, IE11 is technically still a supported browser. So you can argue that it should be supported as a target. You can also argue that MS should fix it.

To argue the above points, I don't think it should be supported as a target because it does not comply with standards. MS deliberately chose to ignore standards or common practice and decided to make things up as they saw fit. The also decided not to implement fixes for those decisions

However MS did actually fix it. They released a new browser. So there really isn't an issue here... except for that fact the people have chosen not to upgrade. Those people will give you plenty of excuses as to why they aren't upgrading or switching, none which are valid.

That 3.25% (other sites say less then 2%) number is also very misleading. In fact most sites list the total IE/Edge ecosystem at right around that 3.25%. 

There can easily be that much error in their results. That result is also a global result. Where are your visitors coming from? What do **your** analytics show? I've built websites for many corporations only to have them tell me 100% of the people accessing them have problems. I show that 80% of the web is Chrome and it works fine in Chrome. Well their user base was all their employees who were forced to use IE7 at the time.

Point being, the % of users using X browser means nothing when not looked at for the user base you are targeting and that of which are actually visiting your site.

So with all that being said, if I needed to support a specific old browser from a modern library... I would fork it and add in the changes myself which I think is the best approach here. 

3 reasons for this...

1. If you ask the author or community to make changes to cover an extremely small percentage of users, and they add those changes, you have no guarantee those changes will stay there. 
2. Someone needs to support those changes. To try and keep those changes supported and functional in the future is a great amount of work.
3. Supporting the old 3% can greatly inhibit the ability to provide a better experience for the 95%+ using modern browsers.

**Comment by amcsi** [comment]
Providing a slightly better experience for some people is no equivalent trade to make the component completely unusable to other people.

3% is not a lot, but not insignificant either. Would you spend effort to be able to potentially get 3% more revenue? Depending on how big the project is, easily.

Also, by now we can easily say that everyone who has a little affinity to technology will take the time to install Chrome or Firefox over IE. The only people remaining are people really oblivious to technology, and, what one must not forget, people working at corporations with company policies limiting installations on their work PCs, only allowing IE to be used; these latter people tend to have not insignificant salaries, and not supporting IE, you lose potential revenue of bored employees happening to browse around and finding what would be an interesting online product and would have otherwise purchased it in the spur of the moment.

Anyway, either way, I'm pretty sure most people using this library are ones using it in proprietary projects for their company. And no matter how you feel IE11 is terrible, should die, and everyone using it should suffer as punishment, the truth is that web development companies tend to require IE11 support in their web projects (because they care about the additional 3% revenue), and not supporting IE11 is a dealbreaker for what library to choose to use. 
I'm actually quite ashamed of not that this project isn't willing to support IE11, but that it was _hiding the fact that it doesn't support it_ by not mentioning it anywhere in the docs. I realized too late that this project didn't, and the work I spent trying to integrate it into the project I was working on at work went to waste. Though this is mainly in my perspective, but frontend devs tend to assume that 3rd party libraries they use support IE11, because all major frameworks do, and almost all libraries do. I really went on a tangent here though.

**Comment by luckyexpert** [comment]
Believe me, I wish IE11 didn’t exist. My life would be much easier. I want to use this on my reservation platform, since this looks like one of the easiest and most intuitive for even the most basic users to figure out. My customers can list their properties, and in turn their customers can make a reservation. But if 3 out of 100 users can’t make a reservation, like @amcsi said, that’s 3% lost revenue for my customers, literally. Any old person still running windows 7 with the bowser that was installed the day they bought their computer will not be able to make a reservation. So for me and probably many others looking for a good date range picker, it has to be all or nothing since we have no control over what browsers our customers use, or for that matter our customer’s customers. It would be different if just some of the features didn’t work on IE, but for it to just do nothing on IE is not acceptable at all. Like I said. This looks like a really good date range picker, it’s definitely the best one for selecting a date range when using 2 separate text boxes. Props to everybody who has contributed to this! This is just my 2 cents on how that 3% really is a big deal for a lot of people.
